payjoin/src/receive/mod.rs

```diff
@@ -113,6 +113,9 @@ impl UncheckedProposal {
         Ok(original_psbt_fee / self.extract_tx_to_schedule_broadcast().weight())
     }
 
+    /// The payjoin protocol version
+    pub fn pj_version(&self) -> usize { self.params.v }
```

Comment:
It seems odd to me to expose this public getter only for the purpose of testing. can you let me know a bit more about your thought process for this design? Why not just expose `pub fn params() -> Params` if exposing it at all?

Reply:
In hindsight I should've kept this PR in draft. It seemed like it _might_ be useful for consumers to be able to get the payjoin version, but I mainly added it here for the purpose of testing. I agree it doesn't make sense to expose it for that purpose only. I see 3 alternatives:

1. Expose `params` like you said. This should probably be avoided unless there's a good reason to do it outside of testing.
2. Write a custom helper in `integration.rs` to get the version - which seems more complicated because the response needs to be decapsulated and decrypted to access the params.
3. Remove the version check altogether in tests. This might be sufficient because the `is_output_substitution_disabled`check implicitly confirms that v=2.

Do you have a preference? I can give (2) a shot.

Reply:
I feel like this PR is actually OK to review and quite thorough, I should have just done `request changes`. I think (3) is ok for our beta software. If there's some clever way to get an is-version-set test in (perhaps unit tests have access to private fields?) that'd be great. If not, I think is_output_substitution_disabled is sufficient to test the important behavior that can't be broken on v1-to-v2 vs v2-to-v2. So I favor (3)